mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -311,7 +324,11 @@ trait Params extends Identifiable with Serializable {
    */
   protected final def set[T](param: Param[T], value: T): this.type = {
     shouldOwn(param)
-    paramMap.put(param.asInstanceOf[Param[Any]], value)
+    if (param.isInstanceOf[StringArrayParam] && value.isInstanceOf[Seq[_]]) {
```

Comment:
How is this going to help with the Python Api though? We don't have ParamPair in Python do we?